sql/plan/showtablestatus.go

```diff
@@ -65,44 +61,60 @@ func (s *ShowTableStatus) Schema() sql.Schema { return showTableStatusSchema }
 
 // RowIter implements the sql.Node interface.
 func (s *ShowTableStatus) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, error) {
-	var rows []sql.Row
-	var tables []string
-	var err error
-	if len(s.Databases) > 0 {
-		for _, db := range s.Catalog.AllDatabases() {
-			if !stringContains(s.Databases, db.Name()) {
-				continue
-			}
+	dbName := s.Database
+	if dbName == "" {
+		return nil, sql.ErrNoDatabaseSelected.New()
+	}
 
-			tables, err = db.GetTableNames(ctx)
+	db, err := s.Catalog.Database(dbName)
+	if err != nil {
+		return nil, err
+	}
 
-			if err != nil {
-				return nil, err
-			}
-		}
-	} else {
-		db, err := s.Catalog.Database(ctx.GetCurrentDatabase())
+	tables, err := db.GetTableNames(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var rows = make([]sql.Row, len(tables))
+
+	for i, tName := range tables {
+		table, _, err := s.Catalog.Table(ctx, s.Database, tName)
 		if err != nil {
 			return nil, err
 		}
 
-		tables, err = db.GetTableNames(ctx)
+		var numRows uint64 = 0
+		var nextAIVal interface{} = nil
+		var dataLength uint64 = 0
+		if st, ok := table.(sql.StatisticsTable); ok {
+			numRows, err = st.NumRows(ctx)
+			if err != nil {
+				return nil, err
+			}
+
+			aiVal, err := st.NextAutoIncrementValue(ctx)
```

Comment:
Like I mentioned earlier, move this out of StatisticsTable and conditionally set this if the table implements AutoIncrementTable